wgsl/syntax.bnf

```diff
@@ -0,0 +1,533 @@
+// This file is not directly compatible with existing BNF parsers. Instead, it
+// uses a dialect for a concise reading experience, such as pattern literals
+// for tokens and generalizing RegEx operations that reduce the need for empty
+// alternatives. For details on interpreting this dialect, see 1.2. Syntax
+// Notation in WebGPU Shading Language (WGSL) Specification.
+
+translation_unit :
+  global_directive * global_decl *
+;
+
+global_directive :
+  diagnostic_directive
+| enable_directive
+| requires_directive
+;
+
+global_decl :
+  ';'
+| global_variable_decl ';'
+| global_value_decl ';'
+| type_alias_decl ';'
+| struct_decl
+| function_decl
+| const_assert_statement ';'
+;
+
+bool_literal :
+  'true'
+| 'false'
+;
```

Comment:
As mentioned in the editor's meeting I'd suggest trying out the following style for aesthetics:
```suggestion
bool_literal
: 'true'
| 'false'
;
```

Reply:
I think this change is good for HTML output but not the source because placing at upper right kind of balances the trailing lower left semicolon. But because rule is boundary terminated in HTML, this is a beautiful way too. We chat a bit on this after near end of meeting this week, so comment of @dneto0 would be appreciated before I apply something soon

Reply:
Heh.  For some reason I prefer the original. It's more readable to me. (My bias?)

Reply:
I think we can make a follow-up issue for the value proposal of `:` in the next line for HTML output, but it is an easy adjustment if momentum builds up for it. In the BNF source, I think I'm also in favor to have at initial line.